plugins/backstage-plugin-coder/docs/guides/sdk-recommendations.md

```diff
@@ -0,0 +1,557 @@
+# Using the Coder SDK
+
+This document walks you through using the Coder SDK from within Backstage.
+
+## Disclaimer
+
+As of May 2024, Coder does not have a fully public, versioned SDK published on a package manager like NPM. Coder does intend to release a true JavaScript/TypeScript SDK, but until that is released, the version exposed through Backstage can be best thought of as a "preview"/"testbed" SDK. It is also used in production for [the official Coder VS Code extension](https://github.com/coder/vscode-coder).
+
+If you encounter any issues while using the Backstage version of the SDK, please don't hesitate to open an issue. We would be happy to get any issues fixed, but expect some growing pains as we collect user feedback.
+
+## Welcome to the Coder SDK!
+
+The Coder SDK for Backstage makes it easy for Backstage admins to bring the entire Coder API into Spotify's Backstage platform. While the Coder Backstage plugin does ship with a collection of ready-made components, those can't meet every user's needs, and so, why not give you access to the full set of building blocks, so you can build a solution tailored to your specific use case?
+
+### Table of contents
+
+This guide covers the following:
+
+- Accessing the SDK from your own custom React components
+  - Accessing the SDK via `useCoderSdk`
+  - Accessing the SDK via `useCoderQuery`
+- Authenticating
+  - The fallback auth UI
+- Performing queries
+  - Recommendations for caching data
+  - How the Coder plugin caches data
+  - The `useCoderQuery` custom hook
+  - Cautions against other common UI caching strategies
+- Performing mutations
```

Comment:
Do we want to create some inner-document links here?

Reply:
Yeah, sorry for not clarifying – definitely planning to add links to everything. Just didn't pay as much attention here, since I wasn't sure if these headers made sense yet